### PR TITLE
[license] Tweak error messages

### PR DESCRIPTION
**Pull request by mbrookes**
I only planned to fix one typo: "use on projects" -> "use in projects", but noticed a few other opportunities.

### CODE REVIEW

File: packages/x-license-pro/src/utils/licenseErrorMessageUtils.ts [hunk 1]
```diff
@@ -14,9 +14,10 @@ function showError(message: string[]) {
 
 export function showInvalidLicenseError() {
   showError([
-    'MUI: Invalid license.',
+    'MUI: Invalid license key.',
     '',
-    'Your license for MUI X is not valid, please visit',
+    'Your MUI X license key isn't valid. Please check',
+    'your license key, or visit
```
Comment (mbrookes) [inline packages/x-license-pro/src/utils/licenseErrorMessageUtils.ts:20]:
```suggestion
    'your license key, or visit',
```

File: packages/x-license-pro/src/utils/licenseErrorMessageUtils.ts [hunk 2]
```diff
@@ -14,9 +14,10 @@ function showError(message: string[]) {
 
 export function showInvalidLicenseError() {
   showError([
-    'MUI: Invalid license.',
+    'MUI: Invalid license key.',
```
Comment (oliviertassinari) [inline packages/x-license-pro/src/utils/licenseErrorMessageUtils.ts:17]: Nice, now it matches with https://mui.com/x/advanced-components/#validation-errors.

**Review by flaviendelangle — APPROVED** [review]

File: packages/x-license-pro/src/utils/licenseErrorMessageUtils.ts [hunk 3]
```diff
@@ -14,9 +14,10 @@ function showError(message: string[]) {
...
     '',
-    'Your license for MUI X is not valid, please visit',
+    'Your MUI X license key isn't valid. Please check',
+    'your license key, or visit',
     'https://mui.com/r/x-get-license to get a valid license.',
```
Comment (mbrookes) [inline packages/x-license-pro/src/utils/licenseErrorMessageUtils.ts:21]:
```suggestion
    'https://mui.com/r/x-get-license to purchase a license.',
```

**Review by oliviertassinari — APPROVED** [review]
I have pushed a few more changes, @mbrookes see if it's OK.

I think that we need to create new error messages to really have a great DX, by being more precise and actionable, but this is already a step forward.

**Review by mbrookes — COMMENTED** [review]
LGTM

**Pull request merged**